core/src/test/scala/integration/kafka/api/CorruptedBrokersTest.scala

```diff
@@ -0,0 +1,117 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package integration.kafka.api
+
+import kafka.api.{IntegrationTestHarness, LeaderAndIsr}
+import kafka.log.Log.offsetFromFile
+import kafka.log.{LogSegment}
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.utils.Time
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+import java.util.Properties
+
+class CorruptedBrokersTest extends IntegrationTestHarness {
+  val brokerCount = 3
+  val controllerId = 0
+  val broker1Id = 1
+  val broker2Id = 2
+  val topic = "topic1"
+  val partition = 0
+  val topicPartition = new TopicPartition(topic, partition)
+
+  serverConfig.setProperty(KafkaConfig.LiDropCorruptedFilesEnableProp, true.toString)
+  serverConfig.setProperty(KafkaConfig.LiLeaderElectionOnCorruptionWaitMsProp, 30000.toString)
+  serverConfig.setProperty(KafkaConfig.ControlledShutdownEnableProp, false.toString)
+  serverConfig.setProperty(KafkaConfig.MinInSyncReplicasProp, 1.toString)
+  serverConfig.setProperty(KafkaConfig.UncleanLeaderElectionEnableProp, true.toString)
+
+  producerConfig.setProperty(ProducerConfig.LINGER_MS_CONFIG, 0.toString)
+  producerConfig.setProperty(ProducerConfig.BATCH_SIZE_CONFIG, 0.toString)
+  producerConfig.setProperty(ProducerConfig.ACKS_CONFIG, "all")
+
+  override def modifyConfigs(props: Seq[Properties]): Unit = {
+    super.modifyConfigs(props)
+    // Make broker 0 controller
+    props.head.setProperty(KafkaConfig.PreferredControllerProp, "true")
+  }
+
+  @Test
+  def isrCleanupOnCorruptedBrokerStartup(): Unit = {
+    val electedControllerId = TestUtils.waitUntilControllerElected(zkClient)
+    val controller = getController().kafkaController
+    assertTrue(electedControllerId == controllerId)
+
+    createTopic(topic, Map(partition -> Seq(broker1Id, broker2Id)))
+
+    val producer = createProducer()
+    sendRecords(producer, 5)
+
+    killBroker(broker1Id)
+    TestUtils.waitUntilTrue(() => zkClient.getBroker(broker1Id).isEmpty, "Broker 1 failed to shut down.")
+
+    killBroker(broker2Id)
+    TestUtils.waitUntilTrue(() => zkClient.getBroker(broker2Id).isEmpty, "Broker 2 failed to shut down.")
+
+    val partitionState = zkClient.getTopicPartitionState(topicPartition)
+    assertTrue(partitionState.exists(state =>
+      state.leaderAndIsr.isr == List(2) &&
+        state.leaderAndIsr.leader == LeaderAndIsr.NoLeader))
+
+    truncateLog(broker1Id, 3)
+    truncateLog(broker2Id, 2)
+
+    restartDeadBroker(broker2Id)
+
+    TestUtils.waitUntilTrue(
+      () => {
+        zkClient.getCorruptedBrokers.get(2).exists(_.clearedFromIsrs)
+      }, "Broker 2 did not get cleaned from ISRs")
+
+    restartDeadBroker(broker1Id)
```

Comment:
Huh, I'm not sure what I was thinking there. Is it possible there was an override in IntegrationTestHarness at one point that since went away?

Anyway, definitely not the case now, so yeah...resolving. Sorry for the noise!